### PR TITLE
[Intel GPU] Add NestedTensorXPU to parseDispatchKey and codegen

### PR DESCRIPTION
**Pull request by min-jean-cho**
Part of https://github.com/intel/torch-xpu-ops/issues/1141.

Add `NestedTensorXPU` dispatch key. 
```
>>> nt = torch.nested.nested_tensor([]).to("xpu")
>>> nt
nested_tensor([

], device='xpu:0')
>>> nt.is_xpu
True
```

### CODE REVIEW

**Comment by EikanWang** [comment]
@min-jean-cho , pls. help fix the issue "[pull / linux-jammy-py3.9-gcc11-mobile-lightweight-dispatch-build / build](https://hud.pytorch.org/pr/pytorch/pytorch/140461#32891769327) ([gh](https://github.com/pytorch/pytorch/actions/runs/11806653123/job/32891769327))
undefined reference to c10::impl::common_device_check_failure(c10::Device, at::Tensor const&, char const*, char const*)'`"

**Comment by EikanWang** [comment]
It is a known issue and we will fix it -

 [xpu / linux-jammy-xpu-py3.9 / test (default, 2, 4, linux.idc.xpu)](https://hud.pytorch.org/pr/pytorch/pytorch/140461#32893654911) ([gh](https://github.com/pytorch/pytorch/actions/runs/11806676677/job/32893654911))
inductor/test_decompose_mem_bound_mm.py::TestDecomposeMemMM::test_check_device

File: aten/src/ATen/native/native_functions.yaml
Comment (fengyuan14) [inline aten/src/ATen/native/native_functions.yaml:1]: We only need definition of `NestedTensorXPU` in PyTorch in-tree at the current stage. As a part of operator implementation, `NestedTensorXPU` is only used in `torch-xpu-ops` `native_functions.yaml` to declare a dispatch for XPU.

Comment (min-jean-cho) [reply aten/src/ATen/native/native_functions.yaml:1]: Thanks, updated.

**Review by guangyey — APPROVED** [review]

**Comment by guangyey** [comment]
Let's refine the title. `NextedTensorXPU` is already existed in PyTorch. Actually, this PR aims to add support for parsing NestedTensorXPU and codegen.

**Comment by min-jean-cho** [comment]
Thanks @guangyey for feedback, updated.

**Review by EikanWang — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**